packages/mixin-get-child-by-name/src/index.ts

```diff
@@ -11,20 +11,36 @@ DisplayObject.prototype.name = null;
 /**
  * Returns the display object in the container.
  *
+ * Recursive searches are done in a preorder traversal.
+ *
  * @method getChildByName
  * @memberof PIXI.Container#
  * @param {string} name - Instance name.
+ * @param {boolean}[isRecursive=false] - Whether to search recursively
  * @return {PIXI.DisplayObject} The child with the specified name.
  */
-Container.prototype.getChildByName = function getChildByName(name: string): DisplayObject
+Container.prototype.getChildByName = function getChildByName(name: string, isRecursive?: boolean): DisplayObject
 {
-    for (let i = 0; i < this.children.length; i++)
+    for (let i = 0, j = this.children.length; i < j; i++)
     {
         if (this.children[i].name === name)
         {
             return this.children[i];
         }
     }
 
+    if (isRecursive)
```

Comment:
This could be more succinct if you put the recursive check inside the first loop. Then you don’t have to have two loops.

Reply:
"I think there’s more test coverage we could be doing for this feature. For instance, if a child and a descendant both have the same name, the highest level in the hierarchy should return first."

You need two loops for this type of traversal (preorder). First you have to check all direct children, before searching one depth below.